server/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorStateManager.java

```diff
@@ -155,11 +155,10 @@ public synchronized void maybeSetState(State proposedState)
       return;
     }
 
-    // if we're trying to switch to a healthy steady state (i.e. RUNNING or SUSPENDED) or IDLE state but haven't had a successful run
+    // if we're trying to switch to a healthy steady state (i.e. RUNNING or SUSPENDED) but haven't had a successful run
     // yet, refuse to switch and prefer the more specific states used for first run (CONNECTING_TO_STREAM,
     // DISCOVERING_INITIAL_TASKS, CREATING_TASKS, etc.)
-    if ((healthySteadyState.equals(proposedState) || BasicState.IDLE.equals(proposedState))
```

Comment:
is this actually necessary?

looking at the logic in markRunFinished it seems like atLeastOneSuccessfulRun should be 1 once we hit this logic

Reply:
`SeekableStreamSupervisor.runInternal` call our newly introduced method even before than the `markRunFinished`